packages/chevrotain/src/scan/lexer_public.ts

```diff
@@ -690,7 +691,11 @@ export class Lexer {
         const errorLine = line
         const errorColumn = column
         let foundResyncPoint = false
-        while (!foundResyncPoint && offset < orgLength) {
+        while (
+          this.config.recoveryEnabled &&
```

Comment:
Suggestion: Due to the high performance nature of the lexing hot loop, please assign `this.config.recoveryEnabled` somewhere locally so it doesn't have to look it up in an object for every loop.

Reply:
You are absolutely correct, apologies for being sloppy. I've added a unit test, and fixed the bug.

Reply:
I chose to do this with a flag to make it more readable, but this could also be done by simply setting the offset to the end of the file. I felt the performance difference would be negligible but am happy to do it that way if you'd prefer. The current method sets the 'length' property of the resulting lexing error to 0 (I interpret this as 0 chars skipped, since we failed), but if setting the offset to the end the value would be the length of unlexed string which may also be a useful metric - let me know what you prefer!

Reply:
I believe just stopping the lexer dead in its tracks is a reasonable solution for this.